frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareAppStoreVpp/SoftwareVppForm/SoftwareVppForm.tsx

```diff
@@ -0,0 +1,298 @@
+import Checkbox from "components/forms/fields/Checkbox";
+import TargetLabelSelector from "components/TargetLabelSelector";
+import SoftwareIcon from "pages/SoftwarePage/components/icons/SoftwareIcon";
+import { CUSTOM_TARGET_OPTIONS } from "pages/SoftwarePage/helpers";
+
+import { generateFormValidation, getUniqueAppId } from "./helpers";
+
+const baseClass = "software-vpp-form";
+
+interface IVppAppListItemProps {
+  app: IVppApp;
+  selected: boolean;
+  uniqueAppId: string;
+  onSelect: (software: IVppApp) => void;
+}
+
+const VppAppListItem = ({
+  app,
+  selected,
+  uniqueAppId,
+  onSelect,
+}: IVppAppListItemProps) => {
+  return (
+    <li className={`${baseClass}__list-item`}>
+      <Radio
+        label={
+          <div className={`${baseClass}__app-info`}>
+            <SoftwareIcon url={app.icon_url} />
+            <span>{app.name}</span>
+          </div>
+        }
+        id={`vppApp-${uniqueAppId}`}
+        checked={selected}
+        value={uniqueAppId}
+        name="vppApp"
+        onChange={() => onSelect(app)}
+      />
+      {app.platform && (
+        <div className="app-platform">
+          {PLATFORM_DISPLAY_NAMES[app.platform]}
+        </div>
+      )}
+    </li>
+  );
+};
+
+interface IVppAppListProps {
+  apps: IVppApp[];
+  selectedApp: IVppApp | null;
+  onSelect: (app: IVppApp) => void;
+}
+
+const VppAppList = ({ apps, selectedApp, onSelect }: IVppAppListProps) => {
+  const uniqueSelectedAppId = selectedApp ? getUniqueAppId(selectedApp) : null;
+  return (
+    <div className={`${baseClass}__list-container`}>
+      <ul className={`${baseClass}__list`}>
+        {apps.map((app) => {
+          const uniqueAppId = getUniqueAppId(app);
+          return (
+            <VppAppListItem
+              key={uniqueAppId}
+              app={app}
+              selected={uniqueSelectedAppId === uniqueAppId}
+              uniqueAppId={uniqueAppId}
+              onSelect={onSelect}
+            />
+          );
+        })}
+      </ul>
+    </div>
+  );
+};
+
+export interface ISoftwareVppFormData {
+  selfService: boolean;
+  targetType: string;
+  customTarget: string;
+  labelTargets: Record<string, boolean>;
+  selectedApp?: IVppApp | null;
+}
+
+export interface IFormValidation {
+  isValid: boolean;
+  customTarget?: { isValid: boolean };
+}
+
+interface ISoftwareVppFormProps {
+  labels: ILabelSummary[] | null;
+  vppApps?: IVppApp[];
+  softwareVppForEdit?: IAppStoreApp;
+  onSubmit: (formData: ISoftwareVppFormData) => void;
+  isLoading?: boolean;
+  onCancel: () => void;
+}
+
+const SoftwareVppForm = ({
+  labels,
+  vppApps,
+  softwareVppForEdit,
+  onSubmit,
+  isLoading = false,
+  onCancel,
+}: ISoftwareVppFormProps) => {
+  const [formData, setFormData] = useState<ISoftwareVppFormData>(
+    softwareVppForEdit
+      ? {
+          selfService: softwareVppForEdit.self_service || false,
+          targetType: "All hosts",
+          customTarget: "labelsIncludeAny",
```

Comment:
@RachelElysia This seems like the thing that was causing the missing selected labels.

Reply:
Yup, fix is in <3